core/src/test/java/io/grpc/PickFirstLoadBalancer2Test.java

```diff
@@ -0,0 +1,231 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.Lists;
+
+import io.grpc.LoadBalancer2.Helper;
+import io.grpc.LoadBalancer2.PickResult;
+import io.grpc.LoadBalancer2.Subchannel;
+import io.grpc.PickFirstBalancerFactory2.Picker;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.net.SocketAddress;
+import java.util.List;
+
+
+/** Unit test for {@link PickFirstBalancerFactory2}. */
+@RunWith(JUnit4.class)
+public class PickFirstLoadBalancer2Test {
+  private LoadBalancer2 loadBalancer;
+  private List<ResolvedServerInfoGroup> servers = Lists.newArrayList();
+  private List<SocketAddress> socketAddresses = Lists.newArrayList();
+  private static Attributes.Key<String> MAJOR_KEY = Attributes.Key.of("major-key");
+  private Attributes affinity = Attributes.newBuilder().set(MAJOR_KEY, "I got the keys").build();
+
+  @Captor
+  private ArgumentCaptor<EquivalentAddressGroup> eagCaptor;
+  @Captor
+  private ArgumentCaptor<Picker> pickerCaptor;
+  @Mock
+  private Helper mockHelper;
+  @Mock
+  private Subchannel mockSubchannel;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    for (int i = 0; i < 3; i++) {
+      SocketAddress addr = new FakeSocketAddress("server" + i);
+      servers.add(ResolvedServerInfoGroup.builder().add(new ResolvedServerInfo(addr)).build());
+      socketAddresses.add(addr);
+    }
+
+    when(mockSubchannel.getAddresses()).thenReturn(new EquivalentAddressGroup(socketAddresses));
+    when(mockHelper.createSubchannel(any(EquivalentAddressGroup.class), any(Attributes.class)))
+        .thenReturn(mockSubchannel);
+
+    loadBalancer = PickFirstBalancerFactory2.getInstance().newLoadBalancer(mockHelper);
+  }
+
+  @Test
+  public void pickAfterResolved() throws Exception {
+    loadBalancer.handleResolvedAddresses(servers, affinity);
+
+    verify(mockHelper).createSubchannel(eagCaptor.capture(), eq(affinity));
+    verify(mockHelper).updatePicker(pickerCaptor.capture());
+
+    assertEquals(new EquivalentAddressGroup(socketAddresses), eagCaptor.getValue());
+    assertEquals(pickerCaptor.getValue().pickSubchannel(affinity, new Metadata()),
```

Comment:
checking my understanding: Is this checking that the two calls of `pickSubChannel` return the same result?